app/models/form_profile.rb

```diff
@@ -77,34 +97,59 @@ def initialize_identity_information(user)
         suffix: user.va_profile.suffix
       },
       date_of_birth: user.birth_date,
-      gender: user.gender
+      gender: user.gender,
+      ssn: user.ssn
     )
   end
 
   def initialize_contact_information(user)
+    address = {
+      street: user.va_profile.address.street,
+      street2: nil,
+      city: user.va_profile.address.city,
+      state: user.va_profile.address.state,
+      postal_code: user.va_profile.address.postal_code,
+      country: user.va_profile.address.country
+    } if user.va_profile.address
     FormContactInformation.new(
-      address: {
-        street: user.va_profile.address.street,
-        street2: nil,
-        city: user.va_profile.address.city,
-        state: user.va_profile.address.state,
-        postal_code: user.va_profile.address.postal_code,
-        country: user.va_profile.address.country
-      },
+      address: address,
       home_phone: user.va_profile.home_phone
     )
   end
 
   def generate_prefill(mappings)
-    mappings.map do |k, v|
+    result = mappings.map do |k, v|
       method_chain = v.map(&:to_sym)
       { k.camelize(:lower) => call_methods(method_chain) }
     end.reduce({}, :merge)
+    clean!(result)
   end
 
   def call_methods(methods)
-    methods.inject(self) { |a, e| a.send e }
+    methods.inject(self) { |a, e| a.send e }.as_json
   rescue NoMethodError
     nil
   end
+
+  def clean!(value)
```

Comment:
I wasn't able to repeat your spec failures, but I simplified this code a bit. I took it from some HCA code to remove unused elements in a submission, but the logic here is somewhat different in that we don't have to worry about stringifying booleans.